src/so_vits_svc_fork/preprocess_hubert_f0.py

```diff
@@ -11,47 +11,51 @@
 from joblib import Parallel, cpu_count, delayed
 from tqdm import tqdm
 
-from . import utils
-from .utils import HUBERT_SAMPLING_RATE
+import utils
+from utils import HUBERT_SAMPLING_RATE
 
 LOG = getLogger(__name__)
 
 
+def _process_one(filepath: Path, hubert_model, sampling_rate: int, hop_length: int,
+                 device: Literal["cuda", "cpu"] = "cuda"):
+    wav, sr = librosa.load(filepath, sr=sampling_rate)
+    soft_path = filepath.parent / (filepath.name + ".soft.pt")
+    if not soft_path.exists():
+        wav16k = librosa.resample(
+            wav, orig_sr=sampling_rate, target_sr=HUBERT_SAMPLING_RATE
+        )
+        wav16k = torch.from_numpy(wav16k).to(device)
+        c = utils.get_hubert_content(hubert_model, wav_16k_tensor=wav16k)
+        torch.save(c.cpu(), soft_path)
+    f0_path = filepath.parent / (filepath.name + ".f0.npy")
+    if not f0_path.exists():
+        f0 = utils.compute_f0_dio(
+            wav, sampling_rate=sampling_rate, hop_length=hop_length
+        )
+        np.save(f0_path, f0)
+
+
+def _process_batch(filepaths: Iterable[Path], sampling_rate: int, hop_length: int, pos: int):
```

Comment:
I am not sure why pulling these functions out to the top level increases performance, but it runs much faster.

Reply:
Like not being able to use LokyBackend?

Reply:
After some quick testing on the memory consumption of the backends with my small/medium dataset:

- multiprocessing backend is the worst. 2 threads will very quickly max out a 3090 and crash it.
- loky will struggle along on but complete with 2 threads. More threads will crash it.
- threading backend is slowest, but best on memory consumption by far.

In all three cases, memory was not getting released after it ran until the python instance shut down. I tried some things to get it to release memory, but didn't have any luck. I'm going to swap in the threading backend, but we should probably make an Issue to track it and fix it so it does not break larger datasets.